### PR TITLE
Update and introduce new workflow

### PR DESCRIPTION
**Pull request by minkimcello**
# Purpose
We created a new workflow and optimized the `publish-preview` workflow. These new updates originated from having to refactor away from `actions/npm` and we saw an opportunity to add new functionality to our automated publishing process.

# Approach
### Publish Release (On Pull Request)
  - What used to be three separate jobs (for checking NPM token, publishing, and generating comment), we combined into one. 
  - Replaced `actions/npm` with our new action [`npm-publish-branch-preview`](https://github.com/minkimcello/actions/tree/mk/npm-publish/npm-publish-branch-preview)
    - With the added functionality of npm tagging with the name of the branch so that when someone submits a PR, they can use their package right away with `npm install microstates@<branch-name>`.
  - Updated token source for comment generator so that Frontside's bot can create the comments from now on. **See TODOs below.**

### Publish Release (On Push)
  - Fixed a bug from the last [PR](https://github.com/microstates/microstates.js/pull/374) by removing -a tag to address [this issue](https://github.com/thefrontside/actions/issues/12).
  - Reverted `ubuntu-16.04` back to `ubuntu-latest`.
    - Using `ubunutu-latest` was causing issues on Github last time but we'll keep an eye on it.

### :star:New:star: Synchronize NPM Tags (On Delete)
  - This workflow is triggered on the default branch when another branch or tag is deleted on Github.
    - `synchronize-npm-tags` will cross check npm tags with branches of the repository to remove unnecessary tags. This action can take in an argument for teams that want to specify which tags to sustain.
      - The `latest` tag is never removed.
      - See [README](https://github.com/minkimcello/actions/tree/mk/npm-publish/synchronize-npm-tags) to see how argument can be passed in.
  - This workflow being triggered from the default branch has the added benefit of preventing unwelcome users from modifying this workflow's behavior.

# TODOs
- [x] Add `POST_COMMENT_TOKEN` to repository secrets from npmjs.
- [x] Approve [pull request #13](https://github.com/thefrontside/actions/pull/13) on `thefrontside/actions`.

# Learning
- The latest changes to `frontside/actions` can be viewed [here](https://github.com/thefrontside/actions/pull/13).
- New revised workflows were tested on a forked repository. You can view the results of the test runs [here](https://github.com/minkimcello/microstates.js/actions).

### CODE REVIEW

File: .github/workflows/publish-preview.yml [hunk 1]
```diff
@@ -7,37 +7,18 @@ on:
...
-    steps:
-    - uses: actions/checkout@v1
-    - name: Post Comment
+    - name: Post Instructions Comment
       uses: thefrontside/actions/post-npm-usage-instructions-comment@master
-      env: 
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
\ No newline at end of file
+      env:
+        GITHUB_TOKEN: ${{ secrets.POST_COMMENT_TOKEN }}
```
Comment (cowboyd) [inline .github/workflows/publish-preview.yml]: Let's call this `GITHUB_POST_COMMENT_TOKEN` so it's completely unambiguous what service this is for

**Comment by cowboyd** [comment]
In the future, when opening PRs that aren't ready to merge because they are gated on other dependencies being merged first, let's open them as a [draft][1] PR along with the Criteria to take them out of the draft status. 

[1]: https://github.blog/2019-02-14-introducing-draft-pull-requests/

File: .github/workflows/publish-preview.yml (hunk 1 above)
Comment (minkimcello) [reply .github/workflows/publish-preview.yml]: Done

**Comment by minkimcello** [comment]
> In the future, when opening PRs that aren't ready to merge because they are gated on other dependencies being merged first, let's open them as a [draft pull request](https://github.blog/2019-02-14-introducing-draft-pull-requests/) along with the Criteria to take them out of the draft status.

@cowboyd Oh okay

**Pull request merged**